### PR TITLE
chore: add a sample app

### LINKED ISSUE #909: Move samples from docs-samples
Hi there, our docs-samples repo still have some [trace samples](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/tree/master/trace). For Node.js, all samples live with their client library if they have one. Can you move the samples over please. Thanks! cc @ofrobots

### PR DESCRIPTION
**Pull request by kjin**
Fixes #909

This sample app was taken directly from https://github.com/GoogleCloudPlatform/nodejs-docs-samples/tree/master/trace. I modified the README (to fix broken links, and also to remove sections that seemed copy-pasted from other READMEs) and package.json (to update the Trace Agent version).

### CODE REVIEW

File: samples/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+1.  Refer to the [this README file][readme] for instructions on
+    running and deploying.
+1.  Install dependencies:
+
+    With `npm`:
+
+        npm install
+
+    or with `yarn`:
```
Comment (JustinBeckwith) [inline samples/README.md]: Let's drop the yarn instructions

File: samples/app.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+const express = require('express');
+const got = require('got');
+
+const app = express();
+const DISCOVERY_URL = 'https://www.googleapis.com/discovery/v1/apis';
+
+// This incoming HTTP request should be captured by Trace
+app.get('/', (req, res) => {
+  // This outgoing HTTP request should be captured by Trace
+  got(DISCOVERY_URL, { json: true })
```
Comment (JustinBeckwith) [inline samples/app.js]: I know it's pretty much a copy, but sprinkling some `async/await` love here would be nice :)

File: samples/app.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START app_yaml]
+runtime: nodejs
+env: flex
```
Comment (JustinBeckwith) [inline samples/app.yaml]: While we're here, is it worth changing this to use standard?

File: samples/package.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "nodejs-docs-samples-trace",
+  "description": "Sample for Google Stackdriver Trace on Google App Engine Flexible Environment.",
+  "version": "0.0.1",
+  "private": true,
+  "license": "Apache-2.0",
+  "author": "Google Inc.",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/GoogleCloudPlatform/nodejs-docs-samples.git"
```
Comment (JustinBeckwith) [inline samples/package.json]: not anymore :)

File: samples/package.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  },
+  "engines": {
+    "node": ">=4.3.2"
```
Comment (JustinBeckwith) [inline samples/package.json]: bump to 8+

File: samples/package.json [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  },
+  "scripts": {
+    "deploy": "gcloud app deploy",
+    "start": "node app.js",
+    "lint": "repo-tools lint",
```
Comment (JustinBeckwith) [inline samples/package.json]: Just drop the lint command pls

File: samples/package.json [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    "pretest": "npm run lint",
```
Comment (JustinBeckwith) [inline samples/package.json]: drop this

File: samples/package.json [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    "e2e-test": "repo-tools test deploy"
+  },
+  "dependencies": {
+    "@google-cloud/trace-agent": "3.3.1",
+    "express": "4.16.2",
+    "got": "8.2.0"
+  },
+  "devDependencies": {
+    "@google-cloud/nodejs-repo-tools": "2.2.1",
+    "semistandard": "^12.0.1"
```
Comment (JustinBeckwith) [inline samples/package.json]: No need for semistandard

File: samples/app.js (hunk 2 above)
Comment (kjin) [reply samples/app.js]: I'd like to make improvements to the sample, but I'm hoping to do that as a separate PR. This one is just for adding it.

File: samples/app.yaml (hunk 3 above)
Comment (kjin) [reply samples/app.yaml]: Done

**Comment by kjin** [comment]
@JustinBeckwith I've made changes, PTAL. Note that `dependencies` are now ranges instead of exact versions.

**Pull request merged**